examples/example_piecewise_target_function.py

```diff
@@ -0,0 +1,162 @@
+"""
+Example for evolutionary regression on a piecewise target function
+==================================================================
+
+Example demonstrating the use of Cartesian genetic programming for
+regression on a piecewise target function by using the conditional (if/else) operator.
+"""
+
+# The docopt str is added explicitly to ensure compatibility with
+# sphinx-gallery.
+docopt_str = """
+   Usage:
+     example_piecewise_target_function.py [--max-generations=<N>]
+
+   Options:
+     -h --help
+     --max-generations=<N>  Maximum number of generations [default: 5000]
+"""
+
+import functools
+
+import matplotlib.pyplot as plt
+import numpy as np
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+
+# %%
+# We define a piecewise target function.  The function applies different
+# transformations to the input depending whether the input is less or greater
+# than zero. Thus to achieve high fitness,
+# an individual must make use of the if/else operator.
+def f_target(x):
+    return np.select([x >= 0, x < 0], [x ** 2 + 1.0, -x])
+
+
+# %%
+# Then we define the objective function for the evolution. It uses the
+# mean-squared error between the output of the expression represented by a given
+# individual and the target function evaluated on a set of pseudo-random points.
+def objective(individual, target_function, rng):
+    """Objective function of the regression task.
+
+    Parameters
+    ----------
+    individual : Individual
+        Individual of the Cartesian Genetic Programming Framework.
+    target_function : Callable
+        Target function.
+    rng: numpy.random.RandomState
+
+    Returns
+    -------
+    Individual
+        Modified individual with updated fitness value.
+    """
+    if individual.fitness is not None:
+        return individual
+
+    n_function_evaluations = 1000
+
+    f = individual.to_func()
+    y = np.empty(n_function_evaluations)
+    x = rng.uniform(-5, 5, size=n_function_evaluations)
+    for i, x_i in enumerate(x):
+        y[i] = f([x_i])[0]
+
+    loss = np.mean((target_function(x) - y) ** 2)
+    individual.fitness = -loss
+
+    return individual
+
+
+# %%
+# Next, we set up the evolutionary search. First, we define the parameters for
+# the population, the genomes of individuals, and the evolutionary
+# algorithm.
+population_params = {
+    "n_parents": 1,
+    "mutation_rate": 0.05,
```

Comment:
```suggestion
    "mutation_rate": 0.03,
```
this makes the example pass after 1590 generations on my machine.